cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml

```diff
@@ -22,7 +22,7 @@ spec:
         name: metadata-agent
         ports:
         - containerPort: 8000
-          hostPort: 8000
+          hostPort: 8799
```

Comment:
Don't you also need to change the logging agent configuration to point to this port?

Reply:
Yes, but the logging agent doesn't support the host as a parameter. We should discuss if the port should be part of the same parameter, or a separate parameter entirely. Making it part of METADATA_AGENT_HOSTNAME would be easier, as we could check for the existence of a ":" in the hostname, and append :8000 if not found. WDYT?

Reply:
I'm confused -- the logging agent uses the `metadata_agent_url` parameter, which can definitely include the port. Are you talking about the environment variables used by the container's entrypoint script? If so, then we can add a METADATA_AGENT_PORT variable or rename the current one to METADATA_AGENT_URL.

Reply:
Yeah I'm referencing the container's entrypoint script.

This is where the sed replace occurs
https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/docker/run.sh#L13

And this is what it's replacing "local-metadata-agent.stackdriver.com":8000, not the :8000. Easy enough to update.
https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/docker/Dockerfile#L9

We can discuss the right naming for the env variable, but for now it's not modifiable in any existing docker image.

Reply:
Are you OK with this being final configuration for Metadata Agent?

Reply:
I'm ok with this configuration for the Metadata agent. However, in order for these changes to matter we still need to make changes to a few other places.

1. The input source for containers needs to incorporate the local_resource_id label. You can see how did it below.
    The section that this needs to be added in is here: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml#L74

```
$ curl -s https://storage.googleapis.com/stackdriver-container-alpha/configs/stackdriver-agents.yaml | cat -n | grep -C 5 local_resource_id

   205	      <match reform.**>
   206	        @type record_reformer
   207	        enable_ruby true
   208	        <record>
   209	          "logging.googleapis.com/local_resource_id" ${"k8s_container.#{tag_parts[4].split('-')[-1]}"}
   210	        </record>
   212	        tag ${if record['stream'] == 'stderr' then 'raw.kubernetes.stderr' else 'raw.kubernetes.stdout' end}
   213	      </match>
```

2. You need to update the google_cloud output plugin for kubernetes.* to enable the metadata agent.

     https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml#L351

    You can do this by adding these lines. However you manage to get these lines in there, you'll need to ensure that the hostname is set to the `spec.nodeName` from the downwards API. We do this in our `run.sh` as you're familiar with here: https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/docker/run.sh#L15

```
# Enable metadata agent lookups.
enable_metadata_agent true
metadata_agent_url "http://local-metadata-agent.stackdriver.com:8799"
```

Reply:
@crassirostris 
Mik, do you agree with these changes? Do you have this in your plans?